src/TestProblem/Hydro/CCSN/Flu_ResetByUser_CCSN.cpp

```diff
@@ -0,0 +1,110 @@
+//-------------------------------------------------------------------------------------------------------
+bool Flu_ResetByUser_Func_CCSN( real fluid[], const double x, const double y, const double z, const double Time,
+                                const double dt, const int lv, double AuxArray[] )
+{
+
+   if ( AuxArray != NULL ) {
+#     ifdef MHD
+      const double Emag = AuxArray[0];
+#     else
+      const double Emag = NULL_REAL;
+#     endif
+
+      fluid[TEMP_IG] = Hydro_Con2Temp( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], fluid[ENGY],
+                                       fluid+NCOMP_FLUID, false, NULL_REAL, Emag,
+                                       EoS_DensEint2Temp_CPUPtr, EoS_AuxArray_Flt, EoS_AuxArray_Int, h_EoS_Table );
+
+      return true;
+   }
+   else
+      return false;
+
+} // FUNCTION : Flu_ResetByUser_CCSN
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Flu_ResetByUser_API_CCSN
+// Description :  API for resetting the fluid array in the CCSN problem
+//
+// Note        :  1. Enabled by the runtime option "OPT__RESET_FLUID"
+//                2. Invoked by EvolveLevel() using the function pointer "Flu_ResetByUser_API_Ptr"
+//                   --> This function pointer is resey by Init_TestProb_Hydro_CCSN()
+//                3. Currently NOT applied to the input uniform array
+//                   --> Init_ByFile() does NOT call this function
+//                4. Currently does not work with "OPT__OVERLAP_MPI"
+//
+// Parameter   :  lv      : Target refinement level
+//                FluSg   : Target fluid sandglass
+//                TimeNew : Current physical time (system has been updated from TimeOld to TimeNew in EvolveLevel())
+//                dt      : Time interval to advance solution (can be different from TimeNew-TimeOld in COMOVING)
+//-------------------------------------------------------------------------------------------------------
+void Flu_ResetByUser_API_CCSN( const int lv, const int FluSg, const double TimeNew, const double dt )
+{
+
+   real fluid[NCOMP_TOTAL];
+   bool Reset;
+
+#  pragma omp parallel for private( fluid ) schedule( runtime )
+   for (int PID=0; PID<amr->NPatchComma[lv][1]; PID++)
+   {
+      for (int k=0; k<PS1; k++)  {
+      for (int j=0; j<PS1; j++)  {
+      for (int i=0; i<PS1; i++)  {
+
+         for (int v=0; v<NCOMP_TOTAL; v++)   fluid[v] = amr->patch[FluSg][lv][PID]->fluid[v][k][j][i];
+
+#           if ( MODEL == HYDRO )
+#           ifdef MHD
+            const real Emag = MHD_GetCellCenteredBEnergyInPatch( lv, PID, i, j, k, amr->MagSg[lv] );
+#           else
+            const real Emag = NULL_REAL;
+#           endif
+#           endif
+
+//          reset temperature initial guess for all cells
+            Reset = Flu_ResetByUser_Func_CCSN( fluid, NULL_REAL, NULL_REAL, NULL_REAL, TimeNew, dt, lv, (double*)&Emag );
```

Comment:
Add an array `AuxArray[1]` to store `Emag`, and replace `(double*)&Emag` with `AuxArray`

Reply:
I replaced it.